packages/esm-appointments-app/src/home-appointments/appointments-list.component.tsx

```diff
@@ -90,17 +87,34 @@ const RenderStatus = ({ status, t, appointmentUuid, mutate }) => {
 };
 
 const AppointmentsBaseTable = () => {
-  const { useBahmniAppointmentsUI: useBahmniUI } = useConfig();
+  const { user } = useSession();
+
+  const { useBahmniAppointmentsUI: useBahmniUI, useFullViewPrivilege, fullViewPrivilege } = useConfig();
   const { isLoading, appointments } = useTodayAppointments();
 
+  const [fullView, setFullView] = useState(false);
+
+  const filteredAppointments = !fullView
+    ? appointments.filter((appointment) => appointment.status === 'Scheduled')
+    : appointments;
+
+  const userHasFullAccess = useFullViewPrivilege && userHasAccess(fullViewPrivilege, user);
+  useEffect(() => {
+    if (!useFullViewPrivilege || userHasFullAccess) {
+      setFullView(true);
+    } else {
+      setFullView(false);
+    }
+  }, []);
```

Comment:
I don't think that this needs to be a useEffect. `const fullView = userHasFullAccess || !useFullViewPrivilege` should be sufficient